### PR TITLE
multiple custom actions support in yaml schema

### PR DESCRIPTION
**Pull request by psacc**
### Description of changes
Allows to configure multiple custom actions, adding support in yaml schema to parse and serialize multiple scripts per event.

Validation logic still untouched, that will be addressed in a different change.

### Tests
* Already covered by existing unit test, added some additional cases to cover the added functionality
* manual cluster creation with multiple custom actions:
```bash
[ec2-user@ip-10-0-0-155 ~]$ /opt/parallelcluster/scripts/fetch_and_run -postinstall
echo hello head OnNodeConfigured 1
echo hello head OnNodeConfigured 2
[ec2-user@ip-10-0-0-155 ~]$ srun sleep 60
[ec2-user@ip-10-0-0-155 ~]$ 
```

### References
* https://github.com/aws/aws-parallelcluster/pull/5098
* https://github.com/aws/aws-parallelcluster-cookbook/pull/1895
* https://github.com/aws/aws-parallelcluster/issues/5022

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [x] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [x] Check if documentation is impacted by this change.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cli/src/pcluster/schemas/cluster_schema.py [hunk 1]
```diff
@@ -1091,42 +1121,22 @@ def make_resource(self, data, **kwargs):
...
-class HeadNodeCustomActionsSchema(BaseSchema):
+class QueueCustomActionsSchema(BaseSchema):
     """Represent the schema for all available custom actions."""
 
-    on_node_start = fields.Nested(HeadNodeCustomActionSchema, metadata={"update_policy": UpdatePolicy.UNSUPPORTED})
-    on_node_configured = fields.Nested(HeadNodeCustomActionSchema, metadata={"update_policy": UpdatePolicy.UNSUPPORTED})
-    on_node_updated = fields.Nested(
-        HeadNodeUpdatableCustomActionSchema, metadata={"update_policy": UpdatePolicy.SUPPORTED}
-    )
+    on_node_start = OneOrManyCustomActionField(metadata={"update_policy": UpdatePolicy.UNSUPPORTED})
```
Comment (eantonin) [inline cli/src/pcluster/schemas/cluster_schema.py]: question, shouldn't this be `UpdatePolicy.QUEUE_UPDATE_STRATEGY`? (also for the field below)

File: cli/src/pcluster/schemas/cluster_schema.py [hunk 2]
```diff
@@ -1067,8 +1067,38 @@ def make_resource(self, data, **kwargs):
...
+        if "Script" in value and "Sequence" in value:
+            raise ValidationError("Both Script and Sequence fields are provided. Only one is allowed.")
+
+        if "Script" in value:
+            return super()._deserialize(value, attr, data, **kwargs)
+
+        if "Sequence" in value:
+            sequence = value["Sequence"]
+            if not isinstance(sequence, list):
+                raise ValidationError("Invalid input type for Sequence, expected list.")
```
Comment (eantonin) [inline cli/src/pcluster/schemas/cluster_schema.py:1106]: question: non-blocking: Adding a unit test for this should be quite easy, why don't you do that?

File: cli/tests/pcluster/schemas/test_cluster_schema/test_scheduler_constraints_for_custom_actions/awsbatch.on_node_configured.yaml [hunk 3]
```diff
@@ -8,10 +8,15 @@ HeadNode:
     KeyName: ec2-key-name
   CustomActions:
     OnNodeConfigured:
-      Script: https://test.tgz  # s3:// | https://
-      Args:
-        - arg1
-        - arg2
+      Sequence:
```
Comment (eantonin) [inline cli/tests/pcluster/schemas/test_cluster_schema/test_scheduler_constraints_for_custom_actions/awsbatch.on_node_configured.yaml:11]: question, non-blocking: I guess you changes this yaml file but not others using custom actions right?

File: cli/src/pcluster/schemas/cluster_schema.py (hunk 1 above)
Comment (psacc) [reply cli/src/pcluster/schemas/cluster_schema.py]: yep, good catch, I'm not very good at cut and paste, let's be DRYer on this one

File: cli/src/pcluster/schemas/cluster_schema.py (hunk 2 above)
Comment (psacc) [reply cli/src/pcluster/schemas/cluster_schema.py:1106]: adding it 😊

File: cli/tests/pcluster/schemas/test_cluster_schema/test_scheduler_constraints_for_custom_actions/awsbatch.on_node_configured.yaml (hunk 3 above)
Comment (psacc) [reply cli/tests/pcluster/schemas/test_cluster_schema/test_scheduler_constraints_for_custom_actions/awsbatch.on_node_configured.yaml:11]: not sure I get what you mean on this one

File: cli/tests/pcluster/config/dummy_cluster_config.py [hunk 4]
```diff
@@ -122,8 +124,22 @@ def dummy_head_node(mocker):
     head_node_imds = Imds(secured=True)
     ssh = Ssh(key_name="test")
 
+    custom_actions = CustomActions(
+        on_node_start=[
+            CustomAction(script="https://tests1", args=["arg1", "arg2"]),
+            CustomAction(script="https://tests2", args=["arg1", "arg2"]),
```
Comment (eantonin) [inline cli/tests/pcluster/config/dummy_cluster_config.py:130]: nitpick, just for the sake of testing various paths, I'll do:
* a test with 2 args
* a test with 1 arg
* a test without args

just changing the parameters of the 3 custom actions you are specifying here

I know you are doing it in `test_cluster_schema,` but still

File: cli/tests/pcluster/schemas/test_cluster_schema/test_scheduler_constraints_for_custom_actions/awsbatch.on_node_configured.yaml (hunk 3 above)
Comment (eantonin) [reply cli/tests/pcluster/schemas/test_cluster_schema/test_scheduler_constraints_for_custom_actions/awsbatch.on_node_configured.yaml:11]: resolved offline

**Review by eantonin — APPROVED** [review]
LGTM

**Pull request merged**